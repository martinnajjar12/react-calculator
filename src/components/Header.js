import React from 'react';
import {
  AppBar,
  Drawer,
  Hidden,
  IconButton,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Toolbar,
  Typography,
  Container,
  Grid,
} from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import HomeRoundedIcon from '@material-ui/icons/HomeRounded';
import FunctionsRoundedIcon from '@material-ui/icons/FunctionsRounded';
import FormatQuoteRoundedIcon from '@material-ui/icons/FormatQuoteRounded';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import { NavLink } from 'react-router-dom';

const useStyles = makeStyles(theme => ({
  root: {
    color: 'inherit',
  },
  menuButton: {
    marginLeft: theme.spacing(2),
    [theme.breakpoints.up('sm')]: {
      display: 'none',
    },
  },
  toolbar: theme.mixins.toolbar,
  drawerPaper: {
    width: '240px',
  },
  logoLink: {
    color: '#ff9e00',
    textDecoration: 'none',
  },

  link: {
    color: '#fff',
    textDecoration: 'none',
  },

  activeLink: {
    fontWeight: 'bolder',
    color: '#ff9e00',
  },
}));

const Header = () => {
  const classes = useStyles();
  const theme = useTheme();
  const [mobileOpen, setMobileOpen] = React.useState(false);

  const handleDrawerToggle = () => {
    setMobileOpen(!mobileOpen);
  };

  const properIcon = text => {
    if (text === 'Home') return (<HomeRoundedIcon />);
    if (text === 'Calculator') return (<FunctionsRoundedIcon />);
    return (<FormatQuoteRoundedIcon />);
  };

  const properPath = text => {
    if (text === 'Home') return '/';
    if (text === 'Calculator') return '/calculator';
    return '/quote';
  };

  const drawer = (
    <div>
      <div className={classes.toolbar} />
      <List>
        {['Home', 'Calculator', 'Quote'].map(text => (
          <NavLink
            key={text}
            exact
            activeClassName={classes.activeLink}
            className={classes.link}
            to={properPath(text)}
            onClick={handleDrawerToggle}
          >
            <ListItem button>
              <ListItemIcon className={classes.root}>{ properIcon(text) }</ListItemIcon>
              <ListItemText primary={text} />
            </ListItem>
          </NavLink>
        ))}
      </List>
    </div>
  );

  const container = window !== undefined ? () => window.document.body : undefined;

  return (
    <div className={classes.root}>
      <AppBar position="sticky" className={classes.appBar}>
        <Toolbar>
          <Container>
            <Grid container justify="space-between" alignItems="center">
              <Typography variant="h5" noWrap className={classes.logoLink}>
                Math Magicians
              </Typography>
              <Hidden xsDown>
                <Grid item container xs={5} md={3} justify="space-between">
                  <NavLink exact activeClassName={classes.activeLink} to="/" className={classes.link}>Home</NavLink>
                  <NavLink exact activeClassName={classes.activeLink} to="/calculator" className={classes.link}>Calculator</NavLink>
                  <NavLink exact activeClassName={classes.activeLink} to="/quote" className={classes.link}>Quote</NavLink>
                </Grid>
              </Hidden>
              <IconButton
                color="inherit"
                aria-label="open drawer"
                edge="start"
                onClick={handleDrawerToggle}
                className={classes.menuButton}
              >
                <MenuIcon />
              </IconButton>
            </Grid>
          </Container>
        </Toolbar>
      </AppBar>
      <nav className={classes.drawer} aria-label="mailbox folders">
        <Hidden smUp>
          <Drawer
            container={container}
            variant="temporary"
            anchor={theme.direction === 'rtl' ? 'left' : 'right'}
            open={mobileOpen}
            onClose={handleDrawerToggle}
            classes={{
              paper: classes.drawerPaper,
            }}
            ModalProps={{
              keepMounted: true,
            }}
          >
            {drawer}
          </Drawer>
        </Hidden>
      </nav>
    </div>
  );
};

export default Header;
