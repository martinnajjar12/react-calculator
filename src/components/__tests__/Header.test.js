// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Header from '../Header';

describe('Header component', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Header />);
    expect(tree).toMatchSnapshot();
  });
});
