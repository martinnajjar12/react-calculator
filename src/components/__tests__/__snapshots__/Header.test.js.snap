// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match the snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
        >
          <h5
            className="MuiTypography-root makeStyles-logoLink-5 MuiTypography-h5 MuiTypography-noWrap"
          >
            Math Magicians
          </h5>
          <button
            aria-label="open drawer"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
  <nav
    aria-label="mailbox folders"
  />
</div>
`;
